### PR TITLE
[DEV-4921] New Budget Function count endpoint

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
New endpoint to provide count of budget functions and budget sub functions for an agency in a specific fiscal year. Similar to #2443, #2448, & #2449.

**Technical details:**
Very similar to the new endpoint added in #2443 after it was modified by #2447 and #2448.
There might be room for SQL optimizations, but after #2447 was merged we realized all performance and API responses can be addressed together after the main functionality is merged.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [ ] Frontend
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [ ] Jira Ticket [DEV-4921](https://federal-spending-transparency.atlassian.net/browse/DEV-4921):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected API (sub-6s)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
No changes to data or database schema
```

### CODE REVIEW

File: usaspending_api/agency/v2/views/budget_function_count.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        return Response(
+            {
+                "toptier_code": self.toptier_code,
+                "fiscal_year": self.fiscal_year,
+                "budget_function_count": len(set([row["budget_function_code"] for row in rows])),
+                "budget_sub_function_count": len(set([row["budget_subfunction_code"] for row in rows])),
+            }
+        )
+
+    def budget_function_queryset(self):
```
Comment (BrianZito) [inline usaspending_api/agency/v2/views/budget_function_count.py:33]: nitpicking, but is there a difference between a query and a queryset? This kind of function was just called `query` before.

Comment (tony-sappe) [reply usaspending_api/agency/v2/views/budget_function_count.py:33]: Before?
Queries are SQL and run in a database. Django ORM has querysets to abstract SQL from the business logic.

Comment (BrianZito) [reply usaspending_api/agency/v2/views/budget_function_count.py:33]: I thought the object class endpoint called it `object_class_query`, but I see that that's not the case. If the return value is a queryset (or could be abstracted to one), then I have no complaints.

Comment (tony-sappe) [reply usaspending_api/agency/v2/views/budget_function_count.py:33]: Thankfully I did code it to return a queryset. Whew, glad I avoided that conflict

**Pull request merged**